Can’t install pijul

**Describe the bug**
Installing `pijul` fails with
```
error: failed to run custom build command for `nettle-sys v1.0.1`

Caused by:
  process didn't exit successfully: `/build/pijul-0.12.0/target/release/build/nettle-sys-7dde18c526a5efcf/build-script-build` (exit code: 101)
--- stdout
cargo:rustc-link-search=native=/nix/store/wc2mfjvrnnzdjazmsf8v0lpif3y9m1a0-nettle-3.5.1/lib
cargo:rustc-link-lib=nettle
cargo:rustc-link-lib=hogweed
cargo:include=/nix/store/nam3ilc4s70985mmb22lb0adbkqlcrvh-nettle-3.5.1-dev/include
cargo:rustc-link-lib=dylib=nettle
cargo:rustc-link-lib=dylib=hogweed
cargo:rustc-link-search=native=/nix/store/wc2mfjvrnnzdjazmsf8v0lpif3y9m1a0-nettle-3.5.1/lib
cargo:rustc-link-lib=dylib=gmp

--- stderr
bindgen-wrapper.h:19:10: fatal error: 'nettle/des-compat.h' file not found
bindgen-wrapper.h:19:10: fatal error: 'nettle/des-compat.h' file not found, err: true
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ()', src/libcore/result.rs:1165:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
**To Reproduce**
`nix-env -iA nixos.pijul`

Output of `nix run nixpkgs.nix-info -c nix-info -m`:
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.84, NixOS, 20.03pre201996.41d921292e9 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(root): `"nixos-20.03pre201996.41d921292e9"`
 - channels(user): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/applications/version-management/pijul/default.nix
+++ b/pkgs/applications/version-management/pijul/default.nix
@@ -1,6 +1,17 @@
 { stdenv, fetchurl, rustPlatform, darwin, openssl, libsodium, nettle, clang, libclang, pkgconfig }:
 
-rustPlatform.buildRustPackage rec {
+let
+  # nettle-sys=1.0.1 requires the des-compat.h header, but it was removed in
+  # nettle 3.5.  See https://nest.pijul.com/pijul_org/pijul/discussions/416
+  # Remove with the next release
+  nettle_34 = nettle.overrideAttrs (_oldAttrs: rec {
+    version = "3.4.1";
+    src = fetchurl {
+      url = "mirror://gnu/nettle/nettle-${version}.tar.gz";
+      sha256 = "1bcji95n1iz9p9vsgdgr26v6s7zhpsxfbjjwpqcihpfd6lawyhgr";
+    };
+  });
+in rustPlatform.buildRustPackage rec {
   pname = "pijul";
   version = "0.12.0";
 
@@ -20,7 +31,7 @@ rustPlatform.buildRustPackage rec {
 
   LIBCLANG_PATH = libclang + "/lib";
 
-  buildInputs = [ openssl libsodium nettle libclang ] ++ stdenv.lib.optionals stdenv.isDarwin
+  buildInputs = [ openssl libsodium nettle_34 libclang ] ++ stdenv.lib.optionals stdenv.isDarwin
     (with darwin.apple_sdk.frameworks; [ CoreServices Security ]);
 
   doCheck = false;

